src/graphql/queries/events.ts

```diff
@@ -37,6 +37,29 @@ export const GET_ALL_EVENTS = gql`
   }
 `;
 
+export const GET_EVENTS_FROM_CACHE = gql`
```

Comment:
Can you please explain why you are essentially "deprecating" `GET_ALL_EVENTS` and switching to `GET_EVENTS_FROM_CACHE`? Isn't `@client` deprecated by Apollo? cc @petorsong 

From my understanding, `GET_ALL_EVENTS` as it exists, will query from the cache, but if the data isn't available in the cache, then it will fetch from the backend automatically. (If you use `useQuery` and NOT `readQuery`).

Reply:
for the sake of getting this fix in, i opted to add back in `@client` since it was working before - @JasonYG is it possible to refactor it this way instead?

Reply:
> pretty straightforward for me - i might have minor gripes with the naming convention of the queries and `useAllEventsFromBackend` but honestly not a big deal

Did you have a naming convention that you preferred? I realized that my variable names here might not be completely consistent with what we have, but I figured that the extra clarity was worth it

Reply:
My bad that comment shouldn't be in this thread^

But I tested out Michelle's fix, and it works. Turns out, it was just the `history.replace` being in the wrong place all along lol.

Thanks for pointing it out though!